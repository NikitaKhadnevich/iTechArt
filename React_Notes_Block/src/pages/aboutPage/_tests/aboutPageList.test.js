/* eslint-disable no-shadow */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-unused-vars */
import React from 'react';
import * as reactRedux from 'react-redux';

import { configure, shallow, mount, create, render } from 'enzyme';
import configureMockStore from 'redux-mock-store'; // ES6 modules
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import AboutPageList from '../aboutPageList';

configure({ adapter: new Adapter() });

describe('About Page', () => {
  const middlewares = [];
  const mockStore = configureMockStore(middlewares);
  const initialState = {
    userData: [
      {
        id: '3',
        title: 'Some Title',
        description: 'Some describtion',
        date: '10/19/2021, 04:19:12 AM',
        sharedTime: '',
      },
    ],
  };
  const store = mockStore(initialState);
  const sharedList = store.getState().userData;
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');

  it('Is it About Page Exist', () => {
    const SharedWrapper = shallow(<AboutPageList />);
    expect(SharedWrapper).toHaveLength(1);
  });
  it('Is it About Page Snap', () => {
    const SharedWrapper = shallow(<AboutPageList />);
    expect(SharedWrapper).toMatchSnapshot();
  });
  it('There is About Page has Title', () => {
    const SharedWrapper = shallow(<AboutPageList />);
    expect(
      SharedWrapper.find('WithStyles(ForwardRef(Typography))').first().text()
    ).toEqual('What am I talking about?');
  });
  it('There is About Page has 2 folders', () => {
    const SharedWrapper = shallow(<AboutPageList />);
    expect(
      SharedWrapper.find('WithStyles(ForwardRef(Typography))')
    ).toHaveLength(2);
  });
  it('are About Folders type are Paragraph', () => {
    const SharedWrapper = shallow(<AboutPageList />);
    expect(
      SharedWrapper.findWhere(
        (n) => n.type() === 'WithStyles(ForwardRef(Typography))'
      )
    );
  });
});
