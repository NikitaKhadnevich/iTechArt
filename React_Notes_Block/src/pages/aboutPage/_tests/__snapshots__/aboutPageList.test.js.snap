// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page Is it About Page Snap 1`] = `
<Styled(WithStyles(ForwardRef(Container)))>
  <Styled(WithStyles(ForwardRef(Container)))>
    <WithStyles(ForwardRef(Typography))
      align="center"
      component="h4"
      variant="h4"
    >
      What am I talking about?
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      align="center"
      paragraph={true}
      variant="h5"
    >
      Nice to see you on this ‘Waha Note’ page. There is a SPA that contains information about the main characters of the ‘Warhammer 40k’ universe. It is very good if you know any information about this Universe, and I hope that this resource will be useful to you. So, this minimalistic list of notes consists of two main pages and one additional page. The first page is titled 'Note' and contains basic information such as name, description, and other information about the current person. The second block is called 'share-note' and this page acts as an archive or collection of interesting notes. Each note can be edited, deleted or sent. In the future I am going to add a filter block, more lines for the current person and maybe a few images, also I want to add user validation. Follow the news and may the will of the Emperor be with you.
    </WithStyles(ForwardRef(Typography))>
  </Styled(WithStyles(ForwardRef(Container)))>
</Styled(WithStyles(ForwardRef(Container)))>
`;
